model/configuration.py

```diff
@@ -769,3 +774,567 @@ def excluded_audio_data_sources(cls, _db):
+        :type value: Any
+        """
+        integration = self._integration_owner.collection_external_integration(db, self._collection_id)
+        ConfigurationSetting.for_externalintegration(
+            setting_name,
+            integration).value = value
+
+    def load(self, db, setting_name):
+        """Loads and returns the library's configuration setting
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param setting_name: Name of the library's configuration setting
+        :type setting_name: string
+
+        :return: Any
+        """
+        integration = self._integration_owner.collection_external_integration(db, self._collection_id)
+        value = ConfigurationSetting.for_externalintegration(
+            setting_name,
+            integration).value
+
+        return value
+
+
+class ConfigurationAttributeType(Enum):
+    """Enumeration of configuration setting types"""
+
+    TEXT = 'text'
+    TEXTAREA = 'textarea'
+    SELECT = 'select'
+    NUMBER = 'number'
+
+    def to_control_type(self):
+        """Converts the value to a attribute type understandable by circulation-web
+
+        :return: String representation of attribute's type
+        :rtype: string
+        """
+        # NOTE: For some reason, circulation-web converts "text" into <text> so we have to turn it into None
+        # In this case circulation-web will use <input>
+        if self.value == self.TEXT.value:
+            return None
+        else:
+            return self.value
+
+
+class ConfigurationAttribute(Enum):
+    """Enumeration of configuration setting attributes"""
+
+    KEY = 'key'
+    LABEL = 'label'
+    DESCRIPTION = 'description'
+    TYPE = 'type'
+    REQUIRED = 'required'
+    DEFAULT = 'default'
+    OPTIONS = 'options'
+    CATEGORY = 'category'
+
+
+class ConfigurationOption(object):
+    """Key-value pair containing information about configuration attribute option"""
+
+    def __init__(self, key, label):
+        """Initializes a new instance of ConfigurationOption class
+
+        :param key: Key
+        :type key: string
+
+        :param label: Label
+        :type label: string
+        """
+        self._key = key
+        self._label = label
+
+    @property
+    def key(self):
+        """Returns option's key
+
+        :return: Option's key
+        :rtype: string
+        """
+        return self._key
+
+    @property
+    def label(self):
+        """Returns option's label
+
+        :return: Option's label
+        :rtype: string
+        """
+        return self._label
+
+    def to_settings(self):
+        """Returns a dictionary containing option metadata in the SETTINGS format
+
+        :return: Dictionary containing option metadata in the SETTINGS format
+        :rtype: Dict
+        """
+        return {
+            'key': self.key,
+            'label': self.label
+        }
+
+
+class ConfigurationMetadataOwner(object):
```

Comment:
I'd prefer something like `HasConfigurationSettings`.

Reply:
Changed it to `HasConfigurationSettings`